grpc/tests/src/test/regression/ee_966.rs

```diff
@@ -0,0 +1,362 @@
+use assert_matches::assert_matches;
+use lazy_static::lazy_static;
+
+use casper_engine_test_support::{
+    internal::{
+        DeployItemBuilder, ExecuteRequestBuilder, InMemoryWasmTestBuilder, UpgradeRequestBuilder,
+        ARG_AMOUNT, DEFAULT_PAYMENT, DEFAULT_PROTOCOL_VERSION, DEFAULT_RUN_GENESIS_REQUEST,
+    },
+    DEFAULT_ACCOUNT_ADDR,
+};
+use casper_execution_engine::{
+    core::{
+        engine_state::{upgrade::ActivationPoint, Error},
+        execution::Error as ExecError,
+    },
+    shared::{
+        host_function_costs::HostFunctionCosts,
+        opcode_costs::OpcodeCosts,
+        storage_costs::StorageCosts,
+        wasm_config::{WasmConfig, DEFAULT_MAX_STACK_HEIGHT, DEFAULT_WASM_MAX_MEMORY},
+    },
+};
+use casper_types::{
+    contracts::DEFAULT_ENTRY_POINT_NAME, runtime_args, ApiError, ProtocolVersion, RuntimeArgs,
+};
+use parity_wasm::builder;
+
+const CONTRACT_EE_966_REGRESSION: &str = "ee_966_regression.wasm";
+const MINIMUM_INITIAL_MEMORY: u32 = 16;
+const DEFAULT_ACTIVATION_POINT: ActivationPoint = 0;
+
+lazy_static! {
+    static ref DOUBLED_WASM_MEMORY_LIMIT: WasmConfig = WasmConfig::new(
+        DEFAULT_WASM_MAX_MEMORY * 2,
+        DEFAULT_MAX_STACK_HEIGHT,
+        OpcodeCosts::default(),
+        StorageCosts::default(),
+        HostFunctionCosts::default()
+    );
+    static ref NEW_PROTOCOL_VERSION: ProtocolVersion = ProtocolVersion::from_parts(
+        DEFAULT_PROTOCOL_VERSION.value().major,
+        DEFAULT_PROTOCOL_VERSION.value().minor,
+        DEFAULT_PROTOCOL_VERSION.value().patch + 1,
+    );
+}
+
+fn make_session_code_with_memory_pages(initial_pages: u32, max_pages: Option<u32>) -> Vec<u8> {
+    let module = builder::module()
+        .function()
+        // A signature with 0 params and no return type
+        .signature()
+        .build()
+        .body()
+        .build()
+        .build()
+        // Export above function
+        .export()
+        .field(DEFAULT_ENTRY_POINT_NAME)
+        .build()
+        // Memory section is mandatory
+        .memory()
+        // Produces entry `(memory (0) initial_pages [max_pages])`
+        .with_min(initial_pages)
+        .with_max(max_pages)
+        .build()
+        .build();
+    parity_wasm::serialize(module).expect("should serialize")
+}
+
+#[ignore]
+#[test]
+fn should_run_ee_966_cant_have_too_small_initial_memory() {
+    let session_code = make_session_code_with_memory_pages(0, Some(0));
+
+    let exec_request_1 = {
```

Comment:
We could remove the `_1` suffix from every test's request, since each only has a single request I think.